staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4867,6 +4867,7 @@ type PodStatusResult struct {
 
 // +genclient
 // +genclient:method=UpdateEphemeralContainers,verb=update,subresource=ephemeralcontainers
+// +genclient:method=Resize,verb=update,subresource=resize
```

Comment:
```suggestion
// +genclient:method=UpdateResize,verb=update,subresource=resize
// +genclient:method=ApplyResize,verb=apply,subresource=resize
// +genclient:method=PatchResize,verb=patch,subresource=resize
```

Since patch and apply are more useful for this feature, and this would be more consistent with what is generated for ephemeralcontainers and status.

Reply:
I don't think we need applypatch, can use the generic patch and pass in a subresource

for apply, there's questions about how well apply will work (see the ResetFields comment on the strategy)

Reply:
Do we have a sense of whether subresources supporting patch really matters?  I guess if you are passing the whole pod here it matters, but for something like /scale - why wouldn't everyone just use PUT?

Reply:
> I don't think we need applypatch, can use the generic patch and pass in a subresource

What is the difference between the generic patch and the patch suggested here ?

> for something like /scale - why wouldn't everyone just use PUT?

I might be wrong here. But can there be resource version mismatch incase of scale subresource?

Reply:
My understanding from a quick chat during an API review session with @liggitt  is that we don't need the "PatchResize" generation and most of our other subresources do not provide (since it can be performed on the root resource by passing a "resize" into the subresource vararg.  This same logic is also be valid for Apply.  So the only change I think I'd still make here is renaming the function to `UpdateResize` for consistency with other subresource update function names.